note_taking-explainer.md

```diff
@@ -0,0 +1,129 @@
+- [touch bar or Siri on OSX](https://support.apple.com/en-au/guide/notes/not9474646a9/mac)
+
+Such integrations could begin to be supported for web apps too, if they had a
+way to identify themselves as note-taking apps and a URL to launch for taking
+new note. It wouldn't cover all the use-cases linked above (eg. voice assistant
+actually adding items to the note) but it covers the core flow of adding a new
+note.
+
+## Proposal
+
+This explainer proposes a new dictionary manifest entry `note_taking` with an
+entry `new_note_url` to specify a URL, within app scope, to launch for taking a
+new note.
+
+```
+{
+  "name": "My note taking app",
+  "start_url": "/index.html",
+  ... other required manifest fields ...
+  "note_taking": { "new_note_url": "/new_note.html" },
+}
+```
+
+The presence of the `note_taking` member signals intent to act as a note-taking
+app, and `new_note_url` is a simple note-taking app capability. The url is
+launched in the same way as the `start_url`, so is affected by the app's other
+properties, such as
+[display mode](https://www.w3.org/TR/appmanifest/#display-member) and
+[link capturing](
+https://github.com/WICG/sw-launch/blob/main/declarative_link_capturing.md).
+
+By using a dictionary we can easily add other note-taking app functionality or
+extend the current functionality, and it keeps the manifest organised with
+related functionality together. It also means note-taking can be specified
+largely orthogonally to other manifest specification changes, which helps to
+keep the manifest specification modular and manageable.
+
+## Alternatives Considered
+
+A common theme of alternatives is to make a generalised interface for web apps
+to advertise their capabilities. Ideally this would let us solve a whole class
+of problems instead of just one case. For example, a similar mechanism to "
+create a new note" could be used to "create new calendar entry" or "add a
+contact".
+
+### Extend the `Shortcuts` Member
+
+If we used the existing [Shortcuts manifest member](
+https://www.w3.org/TR/appmanifest/#shortcuts-member), we would need to add a way
+for the user agent to identify shortcuts as intended to be used for specific
+purposes/tasks (eg. a new optional "purpose" field on each shortcut entry, with
+a defined set of values like "new-note", "new-contact", etc). Some of these
+future tasks might need additional parameters/context, which would further
+extend and complicate the `Shortcuts` member. Additionally, Shortcuts are
+intended to be displayed to the user, while many tasks/capabilities should not
+be displayed in a shortcuts menu, eg. any parameterised or contextual tasks,
+non-task capabilities like [lock screen](https://github.com/WICG/lock-screen)
+support. It seems like extending `Shortcuts` this far would be significantly
+changing the semantics of the member.
+
+### A General Capability Member
+
+There are existing general capabilities approaches for other functionality of
```

Comment:
I'm not exactly sure what the relevance of these four APIs is. They don't seem analogous to "integrations" & I would probably just take it out.

It might be worth mentioning them (or a subset of them) up above by way of precedent for the "manifest fields that have a URL that opens when you do something in the OS". (i.e., if anything, these four are a precedent for the proposed design, not the "integrations" alternative.)

Reply:
To me, "Declarative Web Actions" feels like a more "general integrations" approach. But it also appears to be the one that's least implemented. Hmm, that might be telling us something.

I'll move them up above as background.